File: Backend/TopSys.TopConWeb.Domain/Entities/Lead/Lead.cs
using System;
using System.Collections.Generic;
using TopSys.TopConWeb.Domain.Entities.MotivoPerdas;
using TopSys.TopConWeb.Domain.Entities.Visitas;
using TopSys.TopConWeb.Domain.Enums;
using TopSys.TopConWeb.Domain.Interfaces.Entities;

namespace TopSys.TopConWeb.Domain.Entities.Lead
{
    public class Lead : IHasEndereco
    {
        public virtual Usina Usina { get; set; }
        public int UsinaCodigo { get; set; }
        public int Numero { get; set; }
        public int Ano { get; set; }
        //public Visita Visita { get; set; }
        public int VisitaNumero { get; set; }
        public int VisitaAno { get; set; }
        public int OportunidadeNumero { get; set; }
        public int OportunidadeAno { get; set; }
        public DateTime Data { get; set; }
        public string Cliente { get; set; } = "";
        public int DddTelefone { get; set; }
        public int Telefone { get; set; }
        public int DddCelular { get; set; }
        public int Celular { get; set; }
        public string Email { get; set; } = "";
        public Vendedor Vendedor { get; set; }
        public int? VendedorCodigo { get; set; }
        public CadastroGeral ViaCaptacao { get; set; }
        public int? ViaCaptacaoCodigo { get; set; }
        public LeadFase Fase { get; set; }
        public int? FaseCodigo { get; set; }
        public EClassificacaoTemperatura Classificacao { get; set; }
        public string ProximaEtapa { get; set; } = "";
        public string ObraNome { get; set; } = "";
        public string EnderecoCep { get; set; } = "";
        public string EnderecoLogradouro { get; set; } = "";
        public int EnderecoNumero { get; set; }
        public string EnderecoComplemento { get; set; } = "";
        public string EnderecoBairro { get; set; } = "";
        public int? EnderecoMunicipioCodigo { get; set; } = 0;
        public Municipio EnderecoMunicipio { get; set; }
        public MotivoPerda MotivoPerda { get; set; }
        public int? MotivoPerdaCodigo { get; set; }
        public string ObservacaoInterna { get; set; } = "";
        public LeadContato ContatoPrincipal { get; set; }
        public LeadContato ContatoSecundario { get; set; }
        public string IdCadastro { get; set; } = "";
        public string IdAtualizacao { get; set; } = "";
        public ICollection<LeadLog> Logs { get; set; }

        public string GetClassificacaoDescricao
        {
            get
            {
                if (Classificacao == EClassificacaoTemperatura.Quente)
                    return "QUENTE";
                else if (Classificacao == EClassificacaoTemperatura.Morno)
                    return "MORNO";

                return "FRIO";
            }
        }
    }
}

File: Frontend/TopConWeb-AngularV8/src/app/classes/lead/lead.ts
import { CadastroGeral } from "../cadastro-geral/cadastro-geral";
import { Endereco } from "../endereco/endereco";
import { MotivoPerda } from "../motivo-perda/motivo-perda";
import { Usina } from "../usina/usina";
import { Vendedor } from "../vendedor/vendedor";
import { LeadContato } from "./lead-contato";
import { LeadFase } from "./lead-fase";

export class Fase { codigo: number; color: string; }
export class Classificacao { codigo: number; descricao: string; color: string }

export enum EFaseLead {
    Identificacao = 1,
    Qualificacao = 2,
    Trabalhando = 3,
    Nutrindo = 4,
    Apresentacao = 5,
    Oportunidade = 6,
    FechadoPerdido = 7
}

export enum EClassificacaoLead {
    Quente = 1,
    Morno = 2,
    Frio = 3
}

export const faseLead: Fase[] = [
    { codigo: EFaseLead.Identificacao, color: '#9a9afa' },
    { codigo: EFaseLead.Qualificacao, color: '#9a9afa' },
    { codigo: EFaseLead.Trabalhando, color: '#ffc800' },
    { codigo: EFaseLead.Nutrindo, color: '#ffc800' },
    { codigo: EFaseLead.Apresentacao, color: '#ffc800' },
    { codigo: EFaseLead.Oportunidade, color: '#00a500' },
    { codigo: EFaseLead.FechadoPerdido, color: '#FF8072' }
]

export const classificacaoLead: Classificacao[] = [
    { codigo: EClassificacaoLead.Quente, descricao: 'Quente', color: '#FF8072' },
    { codigo: EClassificacaoLead.Morno, descricao: 'Morno', color: '#ffc800' },
    { codigo: EClassificacaoLead.Frio, descricao: 'Frio', color: '#9a9afa' }
]

export class Lead {
    usina: Usina;
    usinaCodigo: number = 0;
    numero: number = 0;
    ano: number = 0;
    visitaNumero: number = 0;
    visitaAno: number = 0;
    oportunidadeNumero: number = 0;
    oportunidadeAno: number = 0;
    data: Date = new Date();
    cliente: string = '';
    dddTelefone: number = 0;
    telefone: number = 0;
    dddCelular: number = 0;
    celular: number = 0;
    email: string = '';
    vendedor: Vendedor;
    viaCaptacao: CadastroGeral;
    fase: LeadFase = new LeadFase();
    classificacao: number = EClassificacaoLead.Frio;
    proximaEtapa: string;
    obraNome: string;
    endereco: Endereco = new Endereco();
    motivoPerda: MotivoPerda = new MotivoPerda();
    observacaoInterna: string;

    contatoPrincipal: LeadContato = new LeadContato();
    contatoSecundario: LeadContato = new LeadContato();
}